src/weekdays.rs

```diff
@@ -0,0 +1,723 @@
+    /// ```
+    pub fn remove(&mut self, day: Weekday) -> bool {
+        if self.contains(day) {
+            self.0 &= !Self::from_one(day).0;
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Iterate over the `Weekday`s in the collection.
+    ///
+    /// Starting from Monday, in ascending order.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::{Weekday, Weekdays};
+    /// let weekdays = Weekdays::MON | Weekdays::WED | Weekdays::FRI;
+    /// let mut iter = weekdays.iter();
+    /// assert_eq!(iter.next(), Some(Weekday::Mon));
+    /// assert_eq!(iter.next(), Some(Weekday::Wed));
+    /// assert_eq!(iter.next(), Some(Weekday::Fri));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    pub const fn iter(self) -> WeekdaysIter {
+        WeekdaysIter(self)
+    }
+
+    /// Get the first day in the collection, starting from Monday.
+    ///
+    /// Returns `None` if the collection is empty.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::{Weekday, Weekdays};
+    /// assert_eq!(Weekdays::MON.first(), Some(Weekday::Mon));
+    /// assert_eq!(Weekdays::TUE.first(), Some(Weekday::Tue));
+    /// assert_eq!(Weekdays::ALL.first(), Some(Weekday::Mon));
+    /// assert_eq!(Weekdays::EMPTY.first(), None);
+    /// ```
+    pub const fn first(self) -> Option<Weekday> {
+        if self.is_empty() {
+            return None;
+        }
+
+        // Find the first non-zero bit.
+        let bit = 1 << self.0.trailing_zeros();
+
+        Self(bit).single_day()
+    }
+
+    /// Get the last day in the collection, starting from Sunday.
+    ///
+    /// Returns `None` if the collection is empty.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::{Weekday, Weekdays};
+    /// assert_eq!(Weekdays::MON.last(), Some(Weekday::Mon));
+    /// assert_eq!(Weekdays::SUN.last(), Some(Weekday::Sun));
+    /// assert_eq!((Weekdays::MON | Weekdays::TUE).last(), Some(Weekday::Tue));
+    /// assert_eq!(Weekdays::EMPTY.last(), None);
+    /// ```
+    pub fn last(self) -> Option<Weekday> {
+        if self.is_empty() {
+            return None;
+        }
+
+        // Find the last non-zero bit.
+        let bit = 1 << (7 - self.0.leading_zeros());
+
+        Self(bit).single_day()
+    }
+
+    /// Split the collection in two at the given day.
+    ///
+    /// Returns a tuple `(before, after)`. `before` contains all days starting from Monday
+    /// up to but __not__ including `weekday`. `after` contains all days starting from `weekday`
+    /// up to and including Sunday.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::{Weekday, Weekdays};
+    /// let weekdays = Weekdays::MON | Weekdays::WED | Weekdays::FRI;
+    /// let (before, after) = weekdays.split_at(Weekday::Wed);
+    /// assert_eq!(before, Weekdays::MON);
+    /// assert_eq!(after, Weekdays::WED | Weekdays::FRI);
+    /// ```
+    pub const fn split_at(self, weekday: Weekday) -> (Self, Self) {
```

Comment:
Let's drop this.

Reply:
Private method? It is used in `iter_some`.

Reply:
It is used twice in `WeekdaySetIterFrom` and is complex enough that implementing and testing it separately makes sense to me, so I made it private and added a test that covers all possible arguments.